libs/encode.lua

```diff
@@ -150,16 +150,16 @@ local btea = function(packet)
 		stackLen = stackLen + 1
 		packet.stack[stackLen] = 0
 	end
-	packet = byteArray:new(packet.stack) -- Saves resource, instead of using write8
 
 	local chunks, counter = { }, 0
-	while #packet.stack > 0 do
+	while packet:bytesLeft() > 0 do
 		counter = counter + 1
 		chunks[counter] = packet:read32()
 	end
 
 	chunks = xxtea(chunks)
 
+	packet.stack = {}
```

Comment:
The satck must be cleared to add the ciphered data.

Reply:
```Lua
	while #packet.stack > 0 do
		packet:read32()

	chunks = xxtea(chunks)

	packet:write16(#chunks)
```

Clearing the stack here is not needed because all bytes are read, and then it writes the first chunk (which is in position 1)